apps/api/src/tasks/import-transactions.task.spec.ts

```diff
@@ -0,0 +1,403 @@
+import { PrismaService } from '../prisma/prisma.service'
+import { IrisIbanAccountInfo, IrisTransactionInfo } from './dto/response.dto'
+import { ImportTransactionsTask } from './import-transactions.task'
+import { Test, TestingModule } from '@nestjs/testing'
+import { HttpModule } from '@nestjs/axios'
+
+import { MockPrismaService, prismaMock } from '../prisma/prisma-client.mock'
+import { SchedulerRegistry } from '@nestjs/schedule/dist'
+import { DonationsService } from '../donations/donations.service'
+import { ConfigService } from '@nestjs/config'
+import { PersonService } from '../person/person.service'
+import { STRIPE_CLIENT_TOKEN } from '@golevelup/nestjs-stripe'
+import { CampaignService } from '../campaign/campaign.service'
+
+import { VaultService } from '../vault/vault.service'
+import { NotificationModule } from '../sockets/notifications/notification.module'
+import { ExportService } from '../export/export.service'
+import { BankDonationStatus, Campaign, Vault } from '@prisma/client'
+import { toMoney } from '../common/money'
+
+describe('ImportTransactionsTask', () => {
+  let taskService: ImportTransactionsTask
+  let prismaService: PrismaService
```

Comment:
the linter shows prismaService as unused an also there are quite some more linter warnings in this file:

`/Users/igoychev/Projects/podkrepi/api/apps/api/src/tasks/import-transactions.task.spec.ts
   23:7   warning  'prismaService' is assigned a value but never used  @typescript-eslint/no-unused-vars
  150:48  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  214:52  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  217:52  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  220:45  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  225:45  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  229:45  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  233:45  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  237:73  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  353:13  warning  'getIBANSpy' is assigned a value but never used     @typescript-eslint/no-unused-vars
  354:52  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  356:13  warning  'getTrxSpy' is assigned a value but never used      @typescript-eslint/no-unused-vars
  357:52  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  386:13  warning  'getIBANSpy' is assigned a value but never used     @typescript-eslint/no-unused-vars
  387:52  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  389:13  warning  'getTrxSpy' is assigned a value but never used      @typescript-eslint/no-unused-vars
  390:52  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  393:45  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any

/Users/igoychev/Projects/podkrepi/api/apps/api/src/tasks/import-transactions.task.ts
  150:13  warning  'result' is assigned a value but never used  @typescript-eslint/no-unused-vars`

Reply:
I don't seem to get these errors, which are actually quite valid. It's probably some config to make it more strict. That's strange because I'm getting all configs from the project itself, should check that.

Anyway, I've fixed most of the errors which relate to unused variables. One thing to note about the error that repeats the most ` warning Unexpected any. Specify a different type @typescript-eslint/no-explicit-any`. The reason why I've use any as a type (which I naturally avoid) is because I'm trying to spy on methods that are declared as private to the class. This makes it very tricky to test them in jest. After researching this turned out to be one of the few ways. If that is not OK, I will try to find another approach - but this would probably take some hours of experimentation.